frontend/src/metabase/lib/dom.js

```diff
@@ -349,13 +356,35 @@ const getOrigin = url => {
 const getLocation = url => {
   try {
     const { pathname, search, hash } = new URL(url, window.location.origin);
+    const sitePath = getSitePath();
+    const pathNameWithoutSubPath =
+      sitePath === "/"
+        ? pathname
+        : pathname.replace(
+            // Only replace the path from the start to avoid matching the string elsewhere
+            new RegExp(`^${ensureNoTrailingSlash(sitePath)}`),
```

Comment:
I couldn't think of any cleaner alternative yet.

Reply:
This will get busted if the string contains regex-sensitive characters. You could do the following:
```
const pathNameWithoutSubpath = pathName.startsWith(subPath) ? pathName.replace(subPath, "") : pathName;
```